### PR TITLE
update build with a dummy status first to avoid race condition of another watch event while the pod is being created

### LINKED ISSUE #518: duplicate build pods created intermittently
## Expected Behavior
A single pod is created for a given `build` resource

## Actual Behavior
Occasionally multiple pods are created suggesting a timing issue.

## Steps to Reproduce the Problem
Unfortunately it's not easily reproducible however we found the issue when upgrading Jenkins X from knative build `0.1.x` to `0.2.x`.  

## Additional Info
After noodling the code my best guess was there's a timing issue where a build resource is created and a pod is started https://github.com/knative/build/blob/d2a0ec1/pkg/reconciler/build/build.go#L173 meanwhile there's an update to the build resource which results in another `reconcile` being called, as the [build status is nil](https://github.com/knative/build/blob/d2a0ec1/pkg/reconciler/build/build.go#L159) another pod is started.  
I have a PR on the way with a suggested fix which updates a build status before starting the pod, I've deployed this to our own Jenkins X infra and not seen a duplicate pod since so I'm pretty

### PR DESCRIPTION
**Pull request by rawlingsj**
fixes #518

### CODE REVIEW

File: pkg/reconciler/build/build.go [hunk 1]
```diff
@@ -170,6 +170,31 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
...
+			Builder: v1alpha1.ClusterBuildProvider,
+			Cluster: &v1alpha1.ClusterSpec{
+				Namespace: build.Namespace,
+				PodName:   podName,
+			},
+			StartTime: &metav1.Time{
+				Time: time.Now(),
+			},
+		}
+		err = c.updateStatus(build)
```
Comment (shashwathi) [inline pkg/reconciler/build/build.go]: `c.updateStatus(build)` is called twice.

Comment (rawlingsj) [reply pkg/reconciler/build/build.go]: doh thanks - removed the extra line

**Comment by ImJasonH** [comment]
Thanks for finding and fixing this! I wonder if there's some way to test this more conclusively, preferrably with an integration test but I'd even take a real e2e test if it's the only way.

It's a fiddly bug that's dependent on a lot of stuff outside of our control, so I'd really like to be more sure that it's not going to crop back up.

**Comment by scothis** [comment]
I was experiencing the same issue with duplicate build pods. Applying this PR resolved the issue as best as I can tell. Thanks @rawlingsj

**Comment by rawlingsj** [comment]
> I was experiencing the same issue with duplicate build pods. Applying this PR resolved the issue as best as I can tell. 

Great - glad it helped!

@shashwathi thanks for merging :)

**Pull request merged**